lib/trie.js

```diff
@@ -0,0 +1,57 @@
+module.exports = Trie;
+
+function Trie() {
+  this.methods = {}
+}
+
+//insert new nodes to Trie
+Trie.prototype.add = function (path, verb, handler) {
+  var node = this;
+  var parts = getPathParts(path);
+
+  if (parts.length) {
+    for (var i = 0; i < parts.length; i++) {
+      if (!node || !(parts[i] in node)) {
+        node[parts[i]] = new Trie();
+      }
+      if (i + 1 === parts.length) {
+        node[parts[i]].methods = node[parts[i]].methods || {};
+        node[parts[i]].methods[verb] = handler;
+      }
+      node = node[parts[i]];
+    }
+  }
+  else { // if we would ever encounter this situation? 
+    node.methods = node.methods || {};
+    node.methods[verb] = handler;
```

Comment:
@bajtos: I found that for a single `verb+path`, there is only one entry point in the express router. The router creates a stack of handler functions, which only invokes the handler, which was registered first by `router[method](path, handler)`, for the path. Other handlers won't see the light if they're not invoked if `request/response` cycle ends before `next()`.
So, for example, in the code below:
```
router.get('/planets', handlerOne());
router.get('/planets', handlerTwo());
```
only `handlerOne()` will be executed if it sends the `response` and doesn't reach the `next()`.
Thus, I believe, we will need to accommodate this case.
Thus, in the code above, at L19, an Array would be helpful to index registering of handlers for same verb+path:
`  node[parts[i]].methods[verb] = handler.push();`

Your thoughts please!